### ISSUE TITLE
Tutorial example failed on animation

### ISSUE THREAD

**Issue by impvd** [KasperskyLab/Kaspresso#254]
After adding animation in the [tutorial example](https://github.com/KasperskyLab/Kaspresso/tree/master/tutorial), Kaspresso got stuck forever, like:

<img width="500" alt="image" src="https://user-images.githubusercontent.com/37371804/120461850-f66cc400-c3cc-11eb-9056-ce78e36f9dac.png">

`SimpleActivity.kt` as follows:

```kotlin
class SimpleActivity : AppCompatActivity() {
    private lateinit var binding: ActivitySimpleBinding
    ...
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySimpleBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        val animation = AnimationUtils.loadAnimation(this@SimpleActivity, R.anim.slide_up)
        binding.textViewTitle.startAnimation(animation)
    }
```

With `slide_up.xml`:

```xml
<?xml version="1.0" encoding="utf-8"?>
<set xmlns:android="http://schemas.android.com/apk/res/android">
    <scale
        android:duration="5000"
        android:fromXScale="1.0"
        android:fromYScale="1.0"
        android:interpolator="@android:anim/linear_interpolator"
        android:toXScale="1.0"
        android:toYScale="0.0"
        android:repeatCount = "infinite"/>
</set>
```

It sounds like animation is still a [nightmare](https://stackoverflow.com/questions/67738461/android-espresso-onview-stuck-infinitely?noredirect=1#comment119793893_67738461) in Kaspresso, from my understanding, UIautomator could ignore the animations, not sure why it doesn't work in this case.

**Comment by impvd** [comment]
After reading [other examples](https://github.com/KasperskyLab/Kaspresso/blob/master/samples/kautomator-sample/src/androidTest/java/com/kaspersky/kaspresso/kautomatorsample/test/UiSimpleTest.kt), it sounds a lot like we have to separate the UiAutomator case and Espresso case.

At first glance, I suppose we will use Espresso by default, and when we meet animations-like feature, we could switch to UiAutomator automatically.

**Comment by matzuk** [comment]
@impvd Hi!
No, Kaspresso is able to struggle with any animation.
I will research your example today.

**Comment by impvd** [comment]
@matzuk 

Awesome thanks for your help:
https://github.com/impvd/kasp101

**Comment by matzuk** [comment]
@impvd Hi!
I've checked your example. So, the explanation is in the mechanism of Espresso. One of the main ideas of Espresso is to put events on UI to the Main thread through a legal Android mechanism as Handler. It allows keeping consistency on UI without unexpected behavior. Also, it prevents some bunch of flaky tests due to the existence of only one single UI events queue where events are executing one by one.
In your example, there is an inifinite animation that started earlier than UI test actions. It means that Espresso pushed UI events to Handler and these events are waiting in the Handler queue while the animation is executing. 
In a nutshell, it's the right behavior of Espresso.

**Comment by impvd** [comment]
Thanks @matzuk , I see. It sounds like we will have to use UiAutomator in this case.

**Comment by matzuk** [comment]
@impvd Maybe, UiAutomator will work. Just try it.
But, is it a good idea to ignore your animation finish?

**Comment by impvd** [comment]
For sure, just wondering if maybe we could ignore the animation without changing product's source code? @matzuk
